### ISSUE TITLE
Multi binary application.

### ISSUE THREAD

**Issue by adam-cyclones** [nexe/nexe#792]
Hiya I have a question that I can't seem to find if it's been asked before. Is this a horrible anti-pattern:

I have several packages in the style of a mono repo. Each package has its own node_modules. To name my project package names we have: x-dev-server, x-cli and x-lib. Each gets its own nexe binary after being built. None of the binaries do very much on their own but all require a client such as the x-cli to pass args to the other too, let's just say we use child process or net to do this.

Do you feel that this might be heavyweight in terms of performance? If not Is there a way I could just require lib into cli and have dependencies still resolve considering the 3 separate node_modules locations?

**Comment by calebboyd** [comment]
Yeah, unless the x-lib and x-dev-server have reasons (maybe running with systemd?) for being their own binaries I'd just include them in one package. There shouldn't be an issue with resolution, as it will occur to the nearest node_modules folder relative to the file making the request.

**Comment by adam-cyclones** [comment]
Okay that's good, to know.

If I quickly linked to my repo, might make things easier to explain. Because what you have described, isn't quite going to plan for me and I have read other issues but I'm not entirely sure if they cover my case, so here is some real examples:

[http://didi.land](http://didi.land)

In `packages/didi-cli-client`
Currently imports packages/didi-lib

This is a classic client and provider type architecture which aims for independent versioning and dependencies. We have no shared dependencies at the root of the repository.

lib-didi needs X dependencies which live within didi-lib, given I compile a nexe binary of didi-cli-client which imports lib-didi, I cannot resolve a dependcy of lib-didi at runtime.

Do you know which issue I can look at to find the answer, sorry Caleb, and thanks for the teams wonderful work.
